components/Modal/index.js

```diff
@@ -1,11 +1,46 @@
-import PropTypes from 'prop-types';
+import React, { useRef, useCallback, useEffect } from 'react';
+import classNames from './modal.module.css';
 
-const Modal = () => {
-  return <div> </div>;
-};
+export default function Modal({ showModal, setShowModal, render }) {
+  const modalRef = useRef();
+  const closeModal = (e) => {
+    if (modalRef.current === e.target) {
+      setShowModal(false);
+    }
+  };
```

Comment:
I am not understanding why are we using `ref` here. Any specific reason behind it ?

Reply:
@sumitd94 
![Screenshot (361)](https://user-images.githubusercontent.com/70262783/107147100-4208e700-6972-11eb-9b1e-3d8ce5bc2660.png)
 We have given modal background as reference. So whenever modal is open and if we click on modal background (outside area of modal), it gets closed.

Reply:
Instead let's create a backdrop and add a onclick event for closing the modal.

Reply:
@sumitd94 Changes has been made please check.